designs/CentrallyManagingNuGetPackageVersions-VSManagePackages.md

```diff
@@ -0,0 +1,414 @@
+
+#### Before
+
+`$(repo_root)\Directory.Packages.props`
+
+```xml
+<Project>
+  <ItemGroup>
+    <PackageVersion Include="PackageA" Version="1.2.3" />
+    <PackageVersion Include="PackageB" Version="4.5.6" />
+  </ItemGroup>
+</Project>
+```
+
+`$(repo_root)\Project1\Project1.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageA" />
+  </ItemGroup>
+
+</Project>
+```
+
+`$(repo_root)\Project2\Project2.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageB" />
+  </ItemGroup>
+
+</Project>
+```
+
+#### Gesture
+
+The customer updates `PackageA` to version `2.0.0` in Package Manager UI.
+
+#### After
+
+`$(repo_root)\Directory.Packages.props`
+
+```xml
+<Project>
+  <ItemGroup>
+    <PackageVersion Include="PackageA" Version="2.0.0" />
+    <PackageVersion Include="PackageB" Version="4.5.6" />
+  </ItemGroup>
+</Project>
+```
+
+`$(repo_root)\Project1\Project1.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageA" />
+  </ItemGroup>
+
+</Project>
+```
+
+`$(repo_root)\Project2\Project2.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageB" />
+  </ItemGroup>
+
+</Project>
+```
+
+## Requirements
+
+The requirements are defined in the [feature spec](https://github.com/NuGet/Home/wiki/Centrally-managing-NuGet-package-versions).
+
+Note that:
+
+* It's valid for some, not all, projects in the solution to use centrally managed package versions.
+* It's valid for different projects using central package versions to use different `Directory.Packages.props` files.
+* As is valid now, it's valid for some projects to use `packages.config`, while others use `PackageReference`. The `packages.config` projects cannot use centrally managed package versions, and individual `PackageReference` projects may opt in or out of central package version management.
+
+## Solution
+
+A new service should be created. This service will have multiple responsibilities:
```

Comment:
In addition to using ProjectSystem for adding/updating package refs NuGet uses Microsoft.Build to do updates for dotnet scenarios. 
ProjectSystem seems to use Microsoft.Build as well. 
It would be great if this new service can be used across both scenarios - Visual Studio and dotnet ,